plugins/woocommerce/includes/class-wc-download-handler.php

```diff
@@ -776,6 +779,102 @@ public static function track_download( $download, $user_id = null, $user_ip_addr
 			);
 		}
 	}
+
+	/**
+	 * Get secure URL for admin image that works with image src attributes
+	 *
+	 * @param int $attachment_id Attachment ID
+	 * @return string Secure admin image URL
+	 *
+	 * @since x.x.x
+	 */
+	public static function get_admin_image_src_url( $product_id, $attachment_id, $size) {
+
+		if ( ! current_user_can( 'manage_woocommerce' ) ) {
+			return '';
+		}
+
+		return add_query_arg(
+				array(
+					'wc-uploads-image-src'=> $attachment_id,
+					'product'=> $product_id,
+					'nonce' => wp_create_nonce( 'admin-image-src-' . $attachment_id ),
+					// Add size parameter for different image sizes
+					'size' => $size,
+					),
+				trailingslashit( home_url() )
+				);
+	}
+
+	/**
+	 * Serve image directly with appropriate headers for src attributes
+	 *
+	 * @since x.x.x
+	 */
+	public static function serve_admin_image_src() {
+
+		if ( ! isset( $_GET['wc-uploads-image-src'] )
+			|| !isset( $_GET['product'])
+			|| ! isset( $_GET['nonce'] ) ) {
+			return;
+		}
+
+		$attachment_id = absint( $_GET['wc-uploads-image-src'] );
+
+		// Security check
+		if ( ! wp_verify_nonce( $_GET['nonce'], 'admin-image-src-' . $attachment_id )
+				|| ! current_user_can( 'manage_woocommerce' )
+		   ) {
+			self::download_error( __( 'Invalid access token', 'woocommerce' ), '', 403 );
+		}
+
+		// Get file path
+		$file_path = get_attached_file( $attachment_id );
+		/* throw new Exception($file_path, 1); */
+		if ( ! $file_path || ! file_exists( $file_path ) ) {
```

Comment:
Could consider swapping out `file_exists` with `is_readable`, which also checks if the file exists and further confirms it can be read.

```suggestion
		if ( ! $file_path || ! is_readable( $file_path ) ) {
```

Reply:
Thank you, I like the suggestion!